server.js

```diff
@@ -4877,6 +4877,61 @@ cache(function(data, match, sendBadge, request) {
   });
 }));
 
+// Jenkins coverage integration (jacoco)
+camp.route(/^\/jenkins(?:-ci)?\/j\/(http(?:s)?)\/([^/]+)\/(.+)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  var scheme = match[1];  // http(s)
+  var host = match[2];  // example.org:8080
+  var job = match[3];  // folder/job
+  var format = match[4];
+  var options = {
+    json: true,
+    uri: scheme + '://' + host + '/job/' + job
+      + '/lastBuild/jacoco/api/json?tree=instructionCoverage[covered,missed,percentage,total]'
+  };
+  if (job.indexOf('/') > -1 ) {
+    options.uri = scheme + '://' + host + '/' + job
+      + '/lastBuild/jacoco/api/json?tree=instructionCoverage[covered,missed,percentage,total]';
+  }
+
+  if (serverSecrets && serverSecrets.jenkins_user) {
+    options.auth = {
+      user: serverSecrets.jenkins_user,
+      pass: serverSecrets.jenkins_pass
+    };
+  }
+
+  var badgeData = getBadgeData('coverage', data);
+  request(options, function(err, res, json) {
+    if (err !== null) {
```

Comment:
We've now got a slightly more comprehensive `checkErrorResponse` helper function to cover error cases. Quite a lot of the newer badge implementations use it if you want to get an example of how it works.